feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,321 @@
+				verifyDUTTelemetry(t, dut, dutPort, dutConf, dutPeerState, test.lldpEnabled)
+			} else {
+				expOtgLLDPNeighbors := map[string][]lldpNeighbors{
+					"ixia-otg": {},
+				}
+
+				checkOtgLLDPNeighbors(t, otg, otgConfig, expOtgLLDPNeighbors)
+			}
+		})
+	}
+}
+
+// configureDUT configures LLDP on a single node.
+func configureDUT(t *testing.T, name string, lldpEnabled bool) (*ondatra.DUTDevice, *oc.Lldp) {
+	node := ondatra.DUT(t, name)
+	p := node.Port(t, portName)
+	lldp := gnmi.OC().Lldp()
+
+	gnmi.Replace(t, node, lldp.Enabled().Config(), lldpEnabled)
+
+	if lldpEnabled {
+		gnmi.Replace(t, node, lldp.Interface(p.Name()).Enabled().Config(), lldpEnabled)
+	}
+
+	return node, gnmi.GetConfig(t, node, lldp.Config())
+}
+
+func configureATE(t *testing.T, otg *otg.OTG) gosnappi.Config {
+
+	// Device configuration + Ethernet configuration.
+	config := otg.NewConfig(t)
+	srcPort := config.Ports().Add().SetName(portName)
+	srcDev := config.Devices().Add().SetName(ateSrc.Name)
+	srcEth := srcDev.Ethernets().Add().SetName(ateSrc.Name + ".Eth")
+	srcEth.SetPortName(srcPort.Name()).SetMac(ateSrc.MAC)
+
+	// LLDP configuration.
+	lldp := config.Lldp().Add()
+	lldp.SystemName().SetValue(lldpSrc.systemName)
+	lldp.SetName(lldpSrc.otgName)
+	lldp.Connection().SetPortName(portName)
+	lldp.ChassisId().MacAddressSubtype().SetValue(lldpSrc.macAddress)
+
+	// Push config and start protocol.
+	otg.PushConfig(t, config)
+	otg.StartProtocols(t)
+
+	return config
+}
+
+// verifyNodeConfig verifies the config by comparing against the telemetry state object.
+func verifyNodeConfig(t *testing.T, node gnmi.DeviceOrOpts, port *ondatra.Port, conf *oc.Lldp, lldpEnabled bool) {
+	statePath := gnmi.OC().Lldp()
+	state := gnmi.Get(t, node, statePath.State())
+	fptest.LogQuery(t, "Node LLDP", statePath.State(), state)
+
+	if lldpEnabled != state.GetEnabled() {
+		t.Errorf("LLDP enabled got: %t, want: %t.", state.GetEnabled(), lldpEnabled)
+	}
+	if state.GetChassisId() != "" {
+		t.Logf("LLDP ChassisId got: %s", state.GetChassisId())
+	} else {
+		t.Errorf("LLDP chassisID is not proper, got %s", state.GetChassisId())
+	}
+	if state.GetChassisIdType() != 0 {
+		t.Logf("LLDP ChassisIdType got: %s", state.GetChassisIdType())
+	} else {
+		t.Errorf("LLDP chassisIdType is not proper, got %s", state.GetChassisIdType())
+	}
+	if state.GetSystemName() != "" {
+		t.Logf("LLDP SystemName got: %s", state.GetSystemName())
+	} else {
+		t.Errorf("LLDP SystemName is not proper, got %s", state.GetSystemName())
+	}
+
+	got := state.GetInterface(port.Name()).GetName()
+	want := conf.GetInterface(port.Name()).GetName()
+
+	if lldpEnabled && got != want {
+		t.Errorf("LLDP interfaces/interface/state/name = %s, want %s", got, want)
+	}
+}
+
+// checkOtgLLDPMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
+func checkOtgLLDPMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
```

Comment:
OTG in the name should be all caps. See https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Because having two all caps words can be confusing, I would rearrange the words as checkLLDPMetricsOTG.

Reply:
Done